pandas/core/indexes/period.py

```diff
@@ -560,6 +560,19 @@ def is_full(self):
         values = self.values
         return ((values[1:] - values[:-1]) < 2).all()
 
+    @property
+    def freq(self):
+        """Return the frequency object if it is set, otherwise None"""
+        return self._freq
+
+    @freq.setter
+    def freq(self, value):
+        msg = ('Setting PeriodIndex.freq has been deprecated and will be '
+               'removed in a future version; use PeriodIndex.asfreq instead. '
+               'The PeriodIndex.freq setter is not guaranteed to work.')
+        warnings.warn(msg, FutureWarning, stacklevel=2)
```

Comment:
Not sure if we really need to deprecate setting this: it's user facing, but I don't think it really work unless you set to the existing frequency.  Could potentially remove the setter entirely, causing an `AttributeError` to be raised, or could raise a custom `AttributeError` within the setter to give a better message directing users to `.asfreq`.

Reply:
Yes, IMO it was buggy, and as you say, only if it was the same frequency, it was working. But given that corner case, it maybe does not hurt to do the deprecation I would say.

I think the custom AttributeError message is a good idea in the future.